src/_pytest/pathlib.py

```diff
@@ -186,19 +186,20 @@ def cleanup_on_exit(lock_path=lock_path, original_pid=pid):
 
 
 def maybe_delete_a_numbered_dir(path):
-    """removes a numbered directory if its lock can be obtained"""
+    """removes a numbered directory if its lock can be obtained and it does not seem to be in use"""
     try:
         create_cleanup_lock(path)
+        parent = path.parent
```

Comment:
if it succeeded in creating the lock it should remove it

Reply:
Good catch, done.